src/stream.rs

```diff
@@ -367,47 +391,73 @@ pub fn compress_buf(params: &CompressParams,
 pub fn compress_vec(params: &CompressParams,
```

Comment:
This was actually largely only provided because there was a raw binding in the underlying library, but this is actually quite interesting now! It looks like `BrotliEncoderCompressStream` is more low level than `BrotliEncoderWriteData` used above in `Compress`, perhaps that should switch to using this API instead?